solr/webapp/web/js/angular/controllers/query.js

```diff
@@ -169,6 +169,9 @@ solrAdminApp.controller('QueryController',
       purgeParams(params, ["q.alt", "qf", "mm", "pf", "ps", "qs", "tie", "bq", "bf"], $scope.val.defType !== "dismax" && $scope.val.defType !== "edismax");
       purgeParams(params, ["uf", "pf2", "pf3", "ps2", "ps3", "boost", "stopwords", "lowercaseOperators"], $scope.val.defType !== "edismax");
       purgeParams(params, getDependentFields("hl"), $scope.val.hl !== true);
+      purgeParams(params, getDependentFields("hl.tag"), $scope.val["hl.method"] !== "unified" && $scope.val["hl.method"] !== "fastVector");
+      purgeParams(params, getDependentFields("hl.simple"), $scope.val["hl.method"] !== "original");
+      // TODO purge highlighter-implementation-specific params
```

Comment:
For example, we should purge params for `hl.method=unified` and `original` if the user has `fastVector` selected.

Is it okay to hardcode all the names of the fields-to-be-purged (like in line 169-170) or is there a cleverer way? The list of names will be pretty long.